java/bogdana_maksiuta/src/com/zoolatech/lecture3/tasks/_2/UserAccountValidator.java

```diff
@@ -0,0 +1,140 @@
+package com.zoolatech.lecture3.tasks._2;
+
+public class UserAccountValidator implements Validator {
+    IdUserValidator idUserValidator = new IdUserValidator();
+    NameValidator nameValidator = new NameValidator();
+    LastNameValidator lastNameValidator = new LastNameValidator();
+    CountryValidator countryValidator = new CountryValidator();
+    UserEmailValidator userEmailValidator = new UserEmailValidator();
+    PhoneNumberValid phoneNumberValid = new PhoneNumberValid();
+    BirthDay birthDay = new BirthDay();
+    BirthMonth birthMonth = new BirthMonth();
+
+    @Override
+    public boolean isValid(UserAccount userAccount) {
+        boolean isValid = idUserValidator.isValid(userAccount) && nameValidator.isValid(userAccount)
+                && lastNameValidator.isValid(userAccount) && countryValidator.isValid(userAccount)
+                && userEmailValidator.isValid(userAccount) && phoneNumberValid.isValid(userAccount)
+                && birthDay.isValid(userAccount) && birthDay.isValid(userAccount)
+                && birthMonth.isValid(userAccount);
+        if (isValid) {
+            System.out.println("Access granted");
+        } else {
+            System.out.println("Access denied");
+        }
+        return isValid;
+    }
+
+}
+
+class IdUserValidator implements Validator {
+    public boolean isCorrect(UserAccount userAccount) {
+        System.out.println("Validating that 'id' is correct");
+        return userAccount.getUserId() > 0;
+    }
+
+    @Override
+    public boolean isValid(UserAccount userAccount) {
+        return isCorrect(userAccount);
+    }
+}
+
+class NameValidator implements Validator {
```

Comment:
You don't need to define a new validator class for each field - you need to create a validator for each rule type (see task's description) and then pass a field name to an object of this validator.

Reply:
we don't validate "id"? Is there a difference in the implementation of the tasks when all the classes are in 1 file and each file is in different ones? or is it visually better?

Reply:
1. we need to validate id, but in a different manner 
2. usually, it's better to define them in separate files (it's simpler to read code if file doesn't have a lot of it)